src/cpu.jakt

```diff
@@ -1474,13 +1474,38 @@ class CPU {
 
         .s = .x
     }
+    
+    function push_byte(mut this, value: u8) {
+        .system.write_byte(address: (0x100u16 + .s as! u16), value)
+        .s -= 1
+    }
+
+    function push_word(mut this, value: u16) {       
+
+        let first_byte = (value && 0xff00) as! u8
+        .push_byte(value: first_byte)
+
+        let second_byte = (value & 0x00ff) as! u8
+        .push_byte(value: second_byte)
```

Comment:
This core is incorrect. Using logical `&&` and not bit-wise`&` and there's a missing right shift.

To support pushing PC for `JSR`, it should be:
```
let hi_byte = ((value & 0xFF00) >> 8) as! u8;
let lo_byte = (value & 0xFF) as! u8;

.push_byte(value: hi_byte);
.push_byte(value: lo_byte);
```

To avoid a run-time check, it might be preferable to use `as_truncated<u8>()` instead of `as! u8`. @jntrnr, do you have a preference?

Reply:
> This core is incorrect. Using logical && and not bit-wise& and there's a missing right shift.
This was corrected by the fixes commit included as part of the PR. Masking the upper byte is also not needed when rightshifting, as a u16 will have nothing else after the shift anyways. This should also prevent us exceeding the u8 range.